tools/benchmark_tool/openvino/tools/benchmark/main.py

```diff
@@ -295,60 +297,93 @@ def set_throughput_streams():
-                keys = benchmark.ie.get_metric(device, 'SUPPORTED_CONFIG_KEYS')
+                keys = benchmark.core.get_metric(device, 'SUPPORTED_CONFIG_KEYS')
                 logger.info(f'DEVICE: {device}')
                 for k in keys:
                     logger.info(f'  {k}  , {exe_network.get_config(k)}')
 
         # Update number of streams
         for device in device_number_streams.keys():
             key = get_device_type_from_name(device) + '_THROUGHPUT_STREAMS'
-            device_number_streams[device] = benchmark.ie.get_config(device, key)
+            device_number_streams[device] = benchmark.core.get_config(device, key)
 
-        # Number of requests
-        infer_requests = exe_network.requests
-
-        # Iteration limit
-        benchmark.niter = get_number_iterations(benchmark.niter, benchmark.nireq, args.api_type)
-
-        # ------------------------------------ 9. Creating infer requests and filling input blobs ----------------------
+        # ------------------------------------ 9. Creating infer requests and preparing input data ----------------------
         next_step()
 
+        # Create infer requests
+        start_time = datetime.utcnow()
+        requests = benchmark.create_infer_requests(exe_network)
+        duration_ms = f"{(datetime.utcnow() - start_time).total_seconds() * 1000:.2f}"
+        logger.info(f"Create {benchmark.nireq} infer requests took {duration_ms} ms")
+        if statistics:
+                statistics.add_parameters(StatisticsReport.Category.EXECUTION_RESULTS,
+                                          [
+                                              ('create infer requests time (ms)', duration_ms)
+                                          ])
+
+        # Prepare input data
         paths_to_input = list()
         if args.paths_to_input:
             for path in args.paths_to_input:
                 if ":" in next(iter(path), ""):
                     paths_to_input.extend(path)
                 else:
                     paths_to_input.append(os.path.abspath(*path))
-        set_inputs(paths_to_input, batch_size, app_inputs_info, infer_requests)
+
+        data_queue = get_input_data(paths_to_input, app_inputs_info)
+
+        benchmark.latency_groups = get_latency_groups(app_inputs_info)
+
+        if len(benchmark.latency_groups) > 1:
+            logger.info(f"Defined {len(benchmark.latency_groups)} tensor groups:")
+            for group in benchmark.latency_groups:
+                print(f"\t{str(group)}")
+
+        # Iteration limit
+        benchmark.niter = get_number_iterations(benchmark.niter, benchmark.nireq, max(len(info.shapes) for info in app_inputs_info), benchmark.api_type)
+
+        # Set input tensors before first inference
+        for request in requests:
+            request.set_tensors(data_queue.get_next_input())
+        if data_queue.size <= benchmark.nireq: # TODO: Should we enable it if there is no need to set new tensors each time ?
```

Comment:
I think we don't need this hidden switch to legacy behavior. Otherwise, from a user perspective performance can be pretty unexpected depending on the input files number.